test/interop/C/multilocale/EXECENV

```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+# If we don't do this then the dynamic linker will fail to load our library
+# when on Linux.
+unamestr=`uname`
+
+if [[ $unamestr == "Linux" ]]; then
+  echo LD_LIBRARY_PATH=$CHPL_HOME/test/interop/C/multilocale/lib:$LD_LIBRARY_PATH
```

Comment:
The question I'm asking here is why this is needed on Linux but not on Mac. For a second I thought we might have hard-coded the LD path on Mac, but I'm not seeing any `-rpath` flags in the LD invocation.

We do the following when building a shared library on Mac:

```
install_name_tool $(BINNAME) -d "$(BINNAME)"
```

Which just sets the binary name within the library file to be the final file name instead of the TMP we create and copy around.

Reply:
Difference in default linking style?  Macs are dynamic by default

Reply:
My worry here is how the Mac knows where to find the dynamic library when it goes to load it, on Linux obviously you have to explicitly tell `ld` where to look, but it seems like Mac's `dyld` knows it's in the `lib` folder already?